### ISSUE TITLE
Slot content not evaluated for a11y check

### ISSUE THREAD

**Issue by judithhartmann** [open-wc/open-wc#2439]
## Expected behavior

I have this code:

Component:
```ts
@customElement('custom-modal')
export class CustomModal extends LitElement {

  @property({ type: String, attribute: 'aria-labelledby' }) ariaLabelledBy?: string;

  render() {
    return `<div role="dialog" aria-labelledby=${ifDefined(this.ariaLabelledBy)}><slot></slot></div>`
  }
}
```

Test:
```ts
  test('is accessible', async () => {
    const el = await fixture(
      html`<custom-modal aria-labelledby="modal-header"
        ><h1 id="modal-header">Title</h1></custom-modal
      >`,
    );
    expect(el).shadowDom.to.have.descendant('h1');
    await expect(el).to.be.accessible();
  });
  ```
  
  My expected behaviour is, that this test passes, because the modal is correctly labelled through the h1 in the slot

## Actual Behavior

Reference element does not exist, reference elements are empty (the slot is not resolved)
```  
    Error: Accessibility Violations
      ---
      Rule: aria-dialog-name
      Impact: serious
      ARIA dialog and alertdialog nodes should have an accessible name (https://dequeuniversity.com/rules/axe/4.3/aria-dialog-name?application=axeAPI)
      Issue target: custom-modal
      Context: <div  role="dialog" aria-labelledby="modal-header">
                  <slot></slot>
              </div>
      Fix any of the following:
        aria-label attribute does not exist or is empty
        aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
        Element has no title attribute
      ---
        aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  ```

## Additional context

Which packages and versions are you using.

    "@open-wc/lit-helpers": "0.4.1",
    "@open-wc/prettier-config": "1.1.0",
    "@open-wc/testing": "3.1.3",
    "@open-wc/testing-helpers": "2.1.2",
    "@web/dev-server-esbuild": "0.3.0",
    "@web/test-runner": "0.13.27",
     "lit": "2.2.2",

**Comment by Westbrook** [comment]
This is not a question of how the Chai A11y Plugin works or does work as much as the browser simply can't complete attribute references across the shadow DOM boundary, so these two pieces of content will never be associated in the way that you are looking for here.

One option would be to separate the headline content from the copy content:

```js
@customElement('custom-modal')
export class CustomModal extends LitElement {
  render() {
    return `<div role="dialog" aria-labelledby="headline">
        <div id="headline"><slot name="headline"></slot></div>
        <slot></slot>
    </div>`
  }
}
```

Another would be to make the headline a property:

```js
@customElement('custom-modal')
export class CustomModal extends LitElement {

  @property() headline = 'Add Headline Here';

  render() {
    return `<div role="dialog" aria-label=${this.headline}>
        <h1>${this.headline}</h1>
        <slot></slot>
    </div>`
  }
}
```

Or, you could follow some of the suggestions in [this article](https://dev.to/westbrook/testing-accessibility-with-shadow-roots-55cm#pretending-to-be-a-native-element) as the `aria-describedby` and `aria-labelledby` attributes can be addressed similarly.

Finally, you can add your support to [this specification](https://leobalter.github.io/cross-root-aria-delegation/) to help bring this natively to browsers more quickly.

I'm going to close this issue as there's nothing specifically that we can fix in this situation.
